src/components/AddItems.jsx

```diff
@@ -16,14 +16,33 @@ const radioInputOptions = {
 	notSoon: ['notsoon', nextPurchaseDate.notSoon, 'Not Soon'],
 };
 
-export function AddItems() {
+export function AddItems({ data }) {
 	const [listPath] = useStateWithStorage('tcl-shopping-list-path', null);
+	console.log(data);
 
 	const handleSubmit = useCallback(
 		async (event) => {
 			event.preventDefault();
 
 			const itemName = event.target.elements['item-name'].value;
+			const normalizedItemName = itemName
+				.trim()
+				.toLowerCase()
+				.replace(/[&\/\\#, +$!,~%.'":*?<>{}]/g, '');
+			if (data) {
+				const currentItems = data.map((item) =>
+					item.name
```

Comment:
wondering whether useMemo could be a good addition here. would love to hear more opinions!

Reply:
I haven't used it before -- I'll have to look into that a bit later!

Reply:
same here, looking for the first opportunity C: